src/github.ts

```diff
@@ -0,0 +1,140 @@
+import * as github from '@actions/github';
+import * as core from '@actions/core';
+import { components } from '@octokit/openapi-types';
+
+const token = core.getInput('token', { required: true });
+const client = github.getOctokit(token);
+
+/**
+ * Get the current context's PR information
+ */
+export function getPr(): components['schemas']['pull-request-minimal'] {
+  const pullRequest = github.context.payload.workflow_run.pull_requests[0];
+  if (!pullRequest) {
+    core.debug(`NO PR FOUND IN CONTEXT (github.content.payload.workflow_run.pull_requests[0]):`);
+    core.setFailed(`PR is undefined. Exiting.`);
+    core.endGroup(); // Gathering inputs...
+    process.exit(1);
```

Comment:
I would avoid setting an exit code within utils etc. This makes testing more difficult, and also means that exit codes are potentially set outside of the app's entry point, which makes debugging more confusing (you don't know where all the exit codes are set).

Reply:
SGTM! I've swapped this to raise an exception instead, and handle it in `main.ts` in the main try/catch.